src/output.cpp

```diff
@@ -1756,28 +1756,21 @@ get_hp_bar( const int cur_hp, const int max_hp, const bool is_mon )
            ( ratio >  0.0 )            ? strings[10] : strings[11];
 }
 
-std::pair<std::string, nc_color> const &get_light_level( const float light )
+std::pair<std::string, nc_color> get_light_level( const float light )
 {
     using pair_t = std::pair<std::string, nc_color>;
     static std::array<pair_t, 6> const strings {
-        {
-            pair_t {_( "unknown" ), c_pink},
-            pair_t {_( "bright" ), c_yellow},
-            pair_t {_( "cloudy" ), c_white},
-            pair_t {_( "shady" ), c_ltgray},
-            pair_t {_( "dark" ), c_dkgray},
-            pair_t {_( "very dark" ), c_black_white}
-        }
-    };
-
-    const int light_level = ceil( light );
-    if( light_level < 0 ) {
-        return strings[0];
-    } else if( light_level > 5 ) {
-        return strings[5];
-    }
-
-    return strings[light_level];
+         {
+             pair_t {translate_marker( "unknown" ), c_pink},
+             pair_t {translate_marker( "bright" ), c_yellow},
+             pair_t {translate_marker( "cloudy" ), c_white},
+             pair_t {translate_marker( "shady" ), c_ltgray},
+             pair_t {translate_marker( "dark" ), c_dkgray},
+             pair_t {translate_marker( "very dark" ), c_black_white}
+         }
+     };
+    const int light_level = clamp( ( int )ceil( light ), 0, 5 );
```

Comment:
Please try not to use "magic numbers".  `strings.size() - 1` is more robust and clearly states the meaning. Optionally `static_assert(!strings.empty())` can be placed before accessing the array by the clamped index; Also, `( int )` is a C-ish way of casting which is not recommended: hurts type-safety, but kept for compatibility purposes. `static_cast`'s should be used instead. Also, array indexes are essentially `size_t`s, not `int`s.